src/main/kotlin/io/github/mojira/arisa/infrastructure/jira/Functions.kt

```diff
@@ -141,11 +143,26 @@ fun deleteAttachment(context: Lazy<IssueUpdateContext>, attachment: Attachment)
     }
 }
 
-fun createComment(context: Lazy<IssueUpdateContext>, comment: String) {
+fun createComment(
+    context: Lazy<IssueUpdateContext>,
+    comment: String,
+    oldPostedCommentCache: Cache<MutableSet<String>>,
+    newPostedCommentCache: Cache<MutableSet<String>>
+) {
     context.value.otherOperations.add {
         runBlocking {
             Either.catch {
-                context.value.jiraIssue.addComment(comment)
+                val key = context.value.jiraIssue.key
+                val oldPostedComments = oldPostedCommentCache.get(key)
+                val newPostedComments = newPostedCommentCache.getOrAdd(key, mutableSetOf())
+
+                if (oldPostedComments != null && oldPostedComments.contains(comment)) {
```

Comment:
Lets imagine a comment is posted, then next run it isnt. What would happen on the third run? It would be posted or not?

Reply:
Let's assume the comment has been posted in the first run.

- If it is never tried to be posted in the second run, it will be posted in the third run.
- If it tried to be posted but actually not posted because of the preventing-spam mechanism, it will still _not_ be posted in the third run.